models/confirmation.go

```diff
@@ -146,7 +146,18 @@ func (c *Confirmation) ValidateCreatorID(expectedCreatorID string, validationErr
 	return c
 }
 
-func (c *Confirmation) ValidateUserID(expectedUserID string, validationErrors *[]error) *Confirmation {
+// ValidateUserID
+// Validate the user based on UserID and Email (entered as username)
+//
+// expectedUserID: userid of the account
+// expectedEmail: username of the account (email)
+// validationErrors: array of error filled in case of error, empty if everything is ok
+//
+func (c *Confirmation) ValidateUserID(expectedUserID string, expectedEmail string, validationErrors *[]error) *Confirmation {
+
+	if expectedEmail == c.Email && expectedUserID != c.UserId {
```

Comment:
```suggestion
	// if a confirmation was already created for the same email in the past, it is reused for the new request
	if expectedEmail == c.Email && expectedUserID != c.UserId {
```

Reply:
it's not exactly the case. The first invitation is accepted with all the required, but the next ones don't have any UserID, that's the reason one we are forcing to use the one that comes with the email.